### PR TITLE
use table.concat to optimize performance

### PR DESCRIPTION
**Pull request by yxudong**
use table.concat to optimize performance of string concatenation

### CODE REVIEW

**Comment by wu-sheng** [comment]
FYI @membphis

**Comment by wu-sheng** [comment]
Seems testing breaks. Please test.

**Comment by yxudong** [comment]
Sorry, I've fixed it.

File: lib/skywalking/segment_ref.lua [hunk 1]
```diff
@@ -59,14 +59,16 @@ end
...
 function _M.serialize(ref)
-    local encodedRef = '1'
-            .. '-' .. encode_base64(ref.trace_id)
-            .. '-' .. encode_base64(ref.segment_id)
-            .. '-' .. ref.span_id
-            .. '-' .. encode_base64(ref.parent_service)
-            .. '-' .. encode_base64(ref.parent_service_instance)
-            .. '-' .. encode_base64(ref.parent_endpoint)
-            .. '-' .. encode_base64(ref.address_used_at_client)
+    local encodedRef = table.concat({
```
Comment (membphis) [inline lib/skywalking/segment_ref.lua:62]: Performance in this way will be lower. 

`table.concat` is only needed when the number of elements of concat is not fixed.

**Review by membphis — CHANGES_REQUESTED** [review]

File: lib/skywalking/segment_ref.lua (hunk 1 above)
Comment (wu-sheng) [reply lib/skywalking/segment_ref.lua:62]: Thanks for confirming.

**Pull request closed without merging**